File: playground/go/personalization.go
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/personalization"
)

func testPersonalization(ctx context.Context, appID, apiKey string) int {
	personalizationClient, err := personalization.NewClient(appID, apiKey, personalization.US)
	if err != nil {
		panic(err)
	}
	ctx, cancel := context.WithTimeout(ctx, 10*time.Millisecond)
	defer cancel()

	// it will fail expectedly because of the very short timeout to showcase the context usage.
	deleteUserProfileResponse, err := personalizationClient.DeleteUserProfile(ctx, "userToken")
	if err != nil {
		fmt.Printf("request error with DeleteUserProfile: %v\n", err)
		return 1
	}

	printResponse(deleteUserProfileResponse)

	return 0
}

File: playground/javascript/node/personalization.ts
import { ApiError } from '@algolia/client-common';
import { personalizationClient } from '@algolia/client-personalization';

const appId = process.env.ALGOLIA_APPLICATION_ID || '**** APP_ID *****';
const apiKey = process.env.ALGOLIA_RECOMMENDATION_KEY || '**** RECOMMENDATION_API_KEY *****';

// Init client with appId and apiKey
const client = personalizationClient(appId, apiKey, 'eu');

async function testPersonalization() {
  try {
    const res = await client.deleteUserProfile({ userToken: 'userToken' });

    console.log(`[OK]`, res);
  } catch (e) {
    if (e instanceof ApiError) {
      return console.log(`[${e.status}] ${e.message}`, e.stackTrace);
    }

    console.log('[ERROR]', e);
  }
}

testPersonalization();
